pkg/render/node.go

```diff
@@ -1220,6 +1245,27 @@ func (c *nodeComponent) nodeEnvVars() []corev1.EnvVar {
 	if c.bpfDataplaneEnabled() {
 		nodeEnv = append(nodeEnv, corev1.EnvVar{Name: "FELIX_BPFENABLED", Value: "true"})
 	}
+	if c.vppDataplaneEnabled() {
+		nodeEnv = append(nodeEnv, corev1.EnvVar{
+			Name:  "FELIX_USEINTERNALDATAPLANEDRIVER",
+			Value: "false",
+		}, corev1.EnvVar{
+			Name:  "FELIX_DATAPLANEDRIVER",
+			Value: "/usr/local/bin/felix-plugins/felix-api-proxy",
+		}, corev1.EnvVar{
+			Name:  "FELIX_XDPENABLED",
+			Value: "false",
+		})
+		if c.cfg.Installation.KubernetesProvider == operatorv1.ProviderEKS {
+			nodeEnv = append(nodeEnv, corev1.EnvVar{
+				Name:  "FELIX_AWSSRCDSTCHECK",
+				Value: "Disable",
+			}, corev1.EnvVar{
+				Name:  "IP_AUTODETECTION_METHOD",
```

Comment:
IP_AUTODETECTION_METHOD is configurable via the Installation API, so we can't ignore whatever the user has specified there. If VPP requires this, then we should set this in defaulting and validation rather than deep in the render code so that the Installation API is consistent with the rendered objects.